### DISCUSSION TITLE
Any way to get statistics for amount of data sent to each client?

### DISCUSSION THREAD

**Discussion by stodge** [socketio/socket.io#3943] [Q&A] [answered]
I need to track statistics for bandwidth usage, number of clients and hence bandwidth per client. Is there any way to get the amount of data that was sent to a client using `io.emit(...)`?

**Comment by darrachequesne** [comment, marked as answer]
Hmm, there is currently no built-in way to do that. You could listen to the [`packetCreate`](https://github.com/socketio/engine.io/blob/733ad4dc4e551ea5319d0efe8895fce4c79a4752/lib/socket.js#L427) event:

```
socket.conn.on("packetCreate", (packet) => {
  if (packet.data) {
    socket.totalAmount += Buffer.byteLength(packet.data);
  }
});
```

Though it might not represent the actual amount, since the payload can then be compressed [down the line](https://github.com/socketio/engine.io/blob/733ad4dc4e551ea5319d0efe8895fce4c79a4752/lib/transports/polling.js#L292-L302) (it should be OK for WebSocket connections though, if the permessage-deflate extension is disabled, which is the default).
